### PR TITLE
Update, fix UInt.pythonObject, and add tests

### PR DESCRIPTION
**Pull request by turbolent**
- Reduced warnings by applying compiler's recommendation to replace `DictionaryLiteral` with `KeyValuePairs`, see [SE-0214](https://github.com/apple/swift-evolution/blob/master/proposals/0214-DictionaryLiteral.md)
- Cherry-picked @dan-zheng's recent commit "Add `Strideable` conformance, remove slice initializer/ (#22849)"
- Ported the tests for the Python runtime and numpy conversions from the Swift repo to XCTest
- Added the Apache license to the whole repository
- Fixed `UInt.pythonObject` for Python 3. Patch also submitted upstream: https://github.com/apple/swift/pull/23055
- Switched tests to Python 3, removed the tests for unsupported sorting of heterogeneous arrays/lists

### CODE REVIEW

File: Tests/PythonKitTests/NumpyConversionTests.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
+
```
Comment (pvieito) [inline Tests/PythonKitTests/NumpyConversionTests.swift]: Please, remove the blank line.

File: Tests/PythonKitTests/NumpyConversionTests.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import PythonKit
+
+
+class NumpyConversionTests: XCTestCase {
+
+  static var numpyModule: PythonObject?
+
+  override class func setUp() {
+    super.setUp()
+    PythonLibrary.useVersion(3, 7)
```
Comment (pvieito) [inline Tests/PythonKitTests/NumpyConversionTests.swift]: Could we generalize this test to Python 3?

`PythonLibrary.useVersion(3)`

File: Tests/PythonKitTests/PythonRuntimeTests.swift [hunk 3]
```diff
@@ -0,0 +1,284 @@
+
```
Comment (pvieito) [inline Tests/PythonKitTests/PythonRuntimeTests.swift]: Same.

File: Tests/PythonKitTests/PythonRuntimeTests.swift [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+  override class func setUp() {
+    super.setUp()
+    PythonLibrary.useVersion(3, 7)
+  }
+
+  func testCheckVersion() {
+    XCTAssertEqual("3.7.", String(Python.version)!.prefix(4))
+    let versionInfo = Python.versionInfo
+    XCTAssertEqual(3, versionInfo.major)
+    XCTAssertEqual(7, versionInfo.minor)
```
Comment (pvieito) [inline Tests/PythonKitTests/PythonRuntimeTests.swift]: `XCTAssertGreaterThanOrEqual(versionInfo.minor, 0)`

**Comment by pvieito** [comment]
@turbolent Thanks! I have made some comments in the review, mainly about style and to generalize the test for Python 3.

**Review by pvieito — CHANGES_REQUESTED** [review]

File: Tests/PythonKitTests/PythonRuntimeTests.swift (hunk 4 above)
Comment (turbolent) [reply Tests/PythonKitTests/PythonRuntimeTests.swift]: Fixed in 79c5933

File: Tests/PythonKitTests/PythonRuntimeTests.swift (hunk 3 above)
Comment (turbolent) [reply Tests/PythonKitTests/PythonRuntimeTests.swift]: Fixed in 79c5933

File: Tests/PythonKitTests/NumpyConversionTests.swift (hunk 2 above)
Comment (turbolent) [reply Tests/PythonKitTests/NumpyConversionTests.swift]: Fixed in 79c5933

File: Tests/PythonKitTests/NumpyConversionTests.swift (hunk 1 above)
Comment (turbolent) [reply Tests/PythonKitTests/NumpyConversionTests.swift]: Fixed in 79c5933

**Comment by pvieito** [comment]
@turbolent Thanks!

**Pull request merged**